lib/engines/hckinstall/hckinstall.rb

```diff
@@ -271,31 +269,49 @@ def run_second(client:)
     end
 
     def prepare_setup_scripts_config
-      kit_string = @platform['kit']
-      kit_type = kit_string[0..2]
-      kit_version = ''
-      kit_type == 'HCK' || kit_version = kit_string[3..]
+      kit_type, kit_version = parse_kit_info
 
       config = {
         kit_type:,
         hlk_kit_ver: kit_version
       }
 
-      unless @kit_info['download_url'].nil?
+      kit_present = kit_exist?(kit_type, kit_version)
+
+      if @kit_info['download_url'].nil? || kit_present
+        @logger.info("HLK installer #{kit_type}#{kit_version} already exists, download skipped") if kit_present
+        @logger.info('HLK installer download URL is not provided') if @kit_info['download_url'].nil?
+      else
         download_kit_installer(@kit_info['download_url'],
                                "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
       end
 
+      raise(AutoHCKError, "HLK installer #{@kit_path} not found") unless kit_exist?(kit_type, kit_version)
+
+      @logger.info("HLK installer #{kit_type}#{kit_version} was found at #{@kit_path}")
+
+      create_setup_scripts_config(@hck_setup_scripts_path, config)
+    end
+
+    def parse_kit_info
+      kit_string = @platform['kit']
+      kit_type = kit_string[0..2]
+      kit_version = kit_type == 'HCK' ? '' : kit_string[3..]
+      [kit_type, kit_version]
+    end
+
+    def kit_exist?(kit_type, kit_version)
```

Comment:
It looks like it only checks the existence, but it actually sets `@kit_path` and `@kit_is_iso`. I suggest renaming this `find_kit` and returning the path to the kit. `@kit_path` and `@kit_is_iso` should be set in `prepare_setup_scripts_config`.